From b8512b4c6715f0eb7ed0ce4e43ad9ef7ce4feb13 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Thu, 2 Sep 2010 21:33:07 -0700
Subject: [PATCH 08/26] xen/evtchn: make pirq enable/disable unmask/mask

PIRQ interrupts can be controlled with just mask/unmask.

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
(cherry picked from commit f4526f9a78ffb3d3fc9f81636c5b0357fc1beccd)

Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
---
 drivers/xen/events.c |   17 ++++-------------
 1 files changed, 4 insertions(+), 13 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index 34e18a1..7241a20 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -510,15 +510,6 @@ static void shutdown_pirq(unsigned int irq)
 	info->evtchn = 0;
 }
 
-static void enable_pirq(unsigned int irq)
-{
-	startup_pirq(irq);
-}
-
-static void disable_pirq(unsigned int irq)
-{
-}
-
 static void ack_pirq(unsigned int irq)
 {
 	int evtchn = evtchn_from_irq(irq);
@@ -1391,11 +1382,11 @@ static struct irq_chip xen_pirq_chip __read_mostly = {
 	.startup	= startup_pirq,
 	.shutdown	= shutdown_pirq,
 
-	.enable		= enable_pirq,
-	.unmask		= enable_pirq,
+	.enable		= unmask_irq,
+	.unmask		= unmask_irq,
 
-	.disable	= disable_pirq,
-	.mask		= disable_pirq,
+	.disable	= mask_irq,
+	.mask		= mask_irq,
 
 	.eoi		= ack_pirq,
 	.end		= end_pirq,
-- 
1.7.5.4

